_posts/2018-06-19-upgrade-rails-from-5-1-to-5-2.markdown

```diff
@@ -0,0 +1,93 @@
+You can use this task as a guideline as explained thoroughly in
+[this post](http://thomasleecopeland.com/2015/08/06/running-rails-update.html).
+
+As an alternative, check out [RailsDiff](http://railsdiff.org/5.1.6/5.2.0),
+which provides an overview of the changes in a basic Rails app between 5.1.x and
+5.2.x (or any other source/target versions).
+
+<h2 id="application-code">4. Application code</h2>
+
+There were not too many deprecations between 5.1 and 5.2 compared to past
+releases ([see the full changelog here](https://guides.rubyonrails.org/5_2_release_notes.html#upgrading-to-rails-5-2)).
+
+However, one major feature is the introduction of [ActiveStorage](https://edgeguides.rubyonrails.org/active_storage_overview.html).
+
+<h3 id="active-storage">a. Active Storage</h2>
+
+Active Storage is a replacement for [Paperclip](https://rubygems.org/gems/paperclip/)
+and other gems, like [Shrine](https://github.com/shrinerb/shrine). Paperclip in
+particular [has been deprecated](https://github.com/thoughtbot/paperclip#deprecated).
+If you're using Paperclip, the maintainers wrote an in-depth migration guide
+which you can [follow here](https://github.com/thoughtbot/paperclip/blob/master/MIGRATING.md).
+
+<h3 id="credentials">b. Credentials</h2>
+
+Another major feature is the introduction of Credentials. It will eventually
+replace the current `config/secrets.yml`. Since there are also encrypted
+secrets (`config/secrets.yml.enc`), its intention is to clear up the confusion.
+
+Your private credentials will reside in `config/credentials.yml.enc`, a file
+that can be safely committed into your repository (its contents are encrypted).
+To un-encrypt them, there's a master key (`config/master.key`) which should
+*NOT* be in your repository (it's git-ignored by default if you start a new
+Rails 5.2 app). To edit the credentials, you don't directly access
+`credentials.yml.enc`, instead, there's a `bin/rails credentials:edit` task
+which opens an unencrypted version of the file (as long as you have the master
+key).
+
+One controversial aspect of it is the [cross-environment credential sharing](https://github.com/rails/rails/issues/31349),
+the credentials file will keep all of your development, test and production keys.
+For more information on Credentials and a work-around to this last issue, check
+out [this post](https://blog.eq8.eu/til/rails-52-credentials-tricks.html), where
+they use `Hash#dig` to configure per-environment credentials.
+
+<h2 id="next-steps">5. Next steps</h2>
+
+If you successfully followed all of these steps, you should now be running Rails 5.2! Do you have any other useful tips or recommendations? Did we miss anything? Share them with us in the comments section.
+
+If you don't have the time to upgrade your Rails app, check out our [Ruby on Rails
+upgrade](https://fastruby.io) service: [FastRuby.io](https://fastruby.io)
```

Comment:
I think just the link on FastRuby.io is enough. Maybe you don't need it for "Ruby on Rails upgrade"